lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -84,6 +84,23 @@ export default function SubmitGradeForm({
   // Used to handle keyboard input changes for the grade input field.
   const inputRef = useRef<HTMLInputElement | null>(null);
 
+  // Listen for page unloads to make sure instructors do not lose unsaved data
+  useEffect(() => {
+    const listener = (e: BeforeUnloadEvent) => {
+      // If grade was saved or not changed at all, allow to safely exit
+      if (gradeSaved || grade.data === inputRef.current?.value) {
```

Comment:
Currently just checking if the grade value changed.

With the logic from https://github.com/hypothesis/lms/pull/5643, this will need to be extended to also check if the comment changes.